Backend/Controllers/Members.cs

```diff
@@ -0,0 +1,513 @@
+using Microsoft.AspNetCore.Mvc;
+using Backend.Database;
+using Backend.Models;
+using Microsoft.EntityFrameworkCore;
+using Backend.Controllers.DTOs;
+
+namespace Backend.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class Members(PostgresDbContext db) : ControllerBase
+    {
+        // GET: api/activities
+        /// <summary>
+        /// Lists all activities in the database.
+        /// </summary>
+        /// <returns>Said list.</returns>
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Member>>> GetMembers(CancellationToken cancellationToken)
+        {
+            var members = await db.Members
+                .Include(m => m.StudyEnrollments)
+                .ThenInclude(se => se.Study)
+                .Include(m => m.Enrollments)
+                .ToListAsync(cancellationToken);
+
+            var result = members.Select(m => new MemberResponseDTO
+            {
+                Id = m.Id,
+                StudentNumber = m.StudentNumber,
+                FirstName = m.FirstName,
+                LastName = m.LastName,
+                Email = m.Email,
+                PhoneNumber = m.PhoneNumber,
+                Address = m.Address,
+                DateOfBirth = m.DateOfBirth,
+                Notes = m.Notes,
+                RegisteredOn = m.RegisteredOn,
+                PreferredLanguage = m.PreferredLanguage,
+                StudyEnrollments = m.StudyEnrollments.Select(se => new StudyEnrollmentResponseDTO
+                {
+                    Id = se.Id,
+                    StudyId = se.StudyId,
+                    StudyTitle = se.Study.Title,
+                    MemberId = se.MemberId,
+                    MemberName = $"{se.Member.FirstName} {se.Member.LastName}",
+                    EnrollmentDate = se.EnrollmentDate,
+                    CompletionDate = se.CompletionDate,
+                    Status = se.Status
+                }).ToList(),
+                GroupMemberships = db.GroupMemberships
+                    .Where(cm => cm.MemberId == m.Id)
+                    .Include(cm => cm.Group)
+                    .Select(cm => new GroupMembershipResponseDTO
+                    {
+                        Id = cm.Id,
+                        GroupId = cm.GroupId,
+                        GroupName = cm.Group.Name,
+                        GroupType = cm.Group.Type,
+                        MemberId = cm.MemberId,
+                        MemberName = $"{cm.Member.FirstName} {cm.Member.LastName}",
+                        MembershipYear = cm.MembershipYear,
+                        RoleId = cm.Role != null ? cm.Role.Id : (uint?)null,
+                        RoleName = cm.Role != null ? cm.Role.Name : null
+                    }).ToList()
+            });
```

Comment:
Same thing for the other DTO. Do we _have_ to supply all entities as a whole, or can we just refer to the other entities? This code is getting quite indented